evaluator/evaluate_baseline.go

```diff
@@ -0,0 +1,63 @@
+package evaluator
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"time"
+
+	"github.com/RedHatInsights/patchman-clients/vmaas"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+func limitVmaasToBaseline(tx *gorm.DB, system *models.SystemPlatform, vmaasData *vmaas.UpdatesV2Response) error {
+	baseline, err := database.GetBaselineConfig(tx, system)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read system's baseline")
+	}
+	if baseline == nil {
+		// no baseline, nothing to change
+		return nil
+	}
+
+	reportedMap := getReportedAdvisories(vmaasData)
+	reportedNames := make([]string, 0, len(reportedMap))
+	for name := range reportedMap {
+		reportedNames = append(reportedNames, name)
+	}
+
+	var filterOutNames []string
+	err = tx.Model(&models.AdvisoryMetadata{}).Where("name IN (?)", reportedNames).
+		Where("public_date >= ?", baseline.ToTime.Truncate(24*time.Hour)).
+		Pluck("name", &filterOutNames).Error
+	if err != nil {
+		return err
+	}
+
+	// create map of advisories we need to filter out
+	filterOutNamesMap := make(map[string]struct{}, len(filterOutNames))
```

Comment:
It's actually set for us (not map), so I would name it filterOutNamesSet.

Reply:
ok